tfx/components/trainer/fn_args_utils.py

```diff
@@ -86,6 +75,27 @@ def get_common_fn_args(input_dict: Dict[Text, List[types.Artifact]],
   json_format.Parse(exec_properties[constants.TRAIN_ARGS_KEY], train_args)
   json_format.Parse(exec_properties[constants.EVAL_ARGS_KEY], eval_args)
 
+  # Default behavior is train on ‘train’ splits (when splits is empty in train
+  # args) and evaluate on 'eval' splits (when splits is empty in eval args).
+  if not train_args.splits:
+    train_args.splits.append('train')
```

Comment:
just confirm, when it's not set, train_args.splits is [] not None, right?

Reply:
yes, because splits is a repeated string, so the default is [] when splits is not provided by user